sled-agent/src/rack_setup/service.rs

```diff
@@ -264,8 +268,42 @@ impl ServiceInner {
     async fn create_plan(
         &self,
         config: &Config,
-        bootstrap_addrs: impl IntoIterator<Item = Ipv6Addr>,
+        bootstrap_addrs: Vec<Ipv6Addr>,
     ) -> Result<HashMap<SocketAddrV6, SledAllocation>, SetupServiceError> {
+        // Create a rack secret, unless we're in the single-sled case.
+        let mut rack_secret_shares = if bootstrap_addrs.len() > 1 {
+            let total_shares = bootstrap_addrs.len();
+            if config.rack_secret_threshold > 1 {
+                let secret = RackSecret::new();
+                let (shares, verifier) = secret
+                    .split(config.rack_secret_threshold, total_shares)
+                    .map_err(SetupServiceError::SplitRackSecret)?;
+
+                // Sanity check that `split` returned the expected number of
+                // shares (one per bootstrap agent)
+                assert_eq!(shares.len(), total_shares);
+
+                Some(shares.into_iter().map(move |share| ShareDistribution {
+                    threshold: config.rack_secret_threshold,
+                    total_shares,
+                    verifier: verifier.clone(),
+                    share,
+                }))
+            } else {
+                warn!(
+                    self.log,
+                    concat!(
+                        "Skipping rack secret creation due to config",
+                        " (despite discovery of {} bootstrap agents)"
+                    ),
+                    total_shares,
+                );
+                None
+            }
+        } else {
+            None
+        };
```

Comment:
WDYT about splitting this into a separate function so we can add a test?

Reply:
Seems perfectly reasonable; will do.